packages/react-native/React/Fabric/Utils/RCTLinearGradient.mm

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import "RCTLinearGradient.h"
+
+#import <React/RCTConversions.h>
+
+using namespace facebook::react;
+
+@implementation RCTLinearGradient
+
++ (CALayer *)gradientLayerWithSize:(CGSize)size
+                          gradient:(const LinearGradient&) gradient {
+  
+  UIGraphicsImageRenderer *renderer = [[UIGraphicsImageRenderer alloc] initWithSize:size];
+  auto orientation = gradient.orientation;
+  auto colorStops = gradient.colorStops;
+  
+  UIImage *gradientImage = [renderer imageWithActions:^(UIGraphicsImageRendererContext * _Nonnull rendererContext) {
+    CGContextRef context = rendererContext.CGContext;
+    NSMutableArray *colors = [NSMutableArray array];
+    CGFloat locations[colorStops.size()];
+    
+    for (size_t i = 0; i < colorStops.size(); ++i) {
+      const auto &colorStop = colorStops[i];
+      UIColor *color = RCTUIColorFromSharedColor(colorStop.color);
+      [colors addObject:(id)color.CGColor];
+      locations[i] = colorStop.position;
+    }
+    
+    auto colorSpace = getDefaultColorSpace() == ColorSpace::sRGB ? CGColorSpaceCreateDeviceRGB() : CGColorSpaceCreateWithName(kCGColorSpaceDisplayP3);
+    
+    CGGradientRef cgGradient = CGGradientCreateWithColors(colorSpace, (__bridge CFArrayRef)colors, locations);
+    
+    CGPoint startPoint;
+    CGPoint endPoint;
+    
+    if (orientation.type == GradientOrientationType::Angle) {
+      CGFloat angle = std::get<Float>(orientation.value);
+      setPointsFromAngle(angle, size, &startPoint, &endPoint);
+    } else if (orientation.type == GradientOrientationType::Direction) {
+      NSString *direction = [NSString stringWithUTF8String:std::get<std::string>(orientation.value).c_str()];
+      CGFloat angle = getAngleFromDirection(direction, size);
+      setPointsFromAngle(angle, size, &startPoint, &endPoint);
+    } else {
+      // Default to top-to-bottom gradient
+      startPoint = CGPointMake(0.0, 0.0);
+      endPoint = CGPointMake(0.0, size.height);
+    }
+    
+    CGContextDrawLinearGradient(context, cgGradient, startPoint, endPoint, 0);
+    
+    CGColorSpaceRelease(colorSpace);
+    CGGradientRelease(cgGradient);
+  }];
+  
+  CALayer *gradientLayer = [CALayer layer];
+  gradientLayer.contents = (__bridge id)gradientImage.CGImage;
+  
+  return gradientLayer;
+}
+
+
+// Spec: https://www.w3.org/TR/css-images-3/#linear-gradient-syntax
+// Reference: https://github.com/chromium/chromium/blob/d32abbe13f5d52be7127fe25d5b778498165fab8/third_party/blink/renderer/core/css/css_gradient_value.cc#L1057
+static void setPointsFromAngle(CGFloat angle, CGSize size, CGPoint *startPoint, CGPoint *endPoint) {
```

Comment:
nit: instead of out params, could this return a struct/tuple? Caller could then destructure (in IIFE if needed)